src/gui/plugins/visualize_frustum/VisualizeFrustum.cc

```diff
@@ -0,0 +1,442 @@
+  {
+    /// \brief Transport node
+    public: transport::Node node;
+
+    /// \brief Scene Pointer
+    public: rendering::ScenePtr scene;
+
+    /// \brief Pointer to FrustumVisual
+    public: rendering::FrustumVisualPtr frustum;
+
+    /// \brief URI sequence to the frustum link
+    public: std::string frustumString{""};
+
+    /// \brief LaserScan message from sensor
+    public: msgs::LogicalCameraSensor msg;
+
+    /// \brief Pose of the frustum visual
+    public: math::Pose3d frustumPose{math::Pose3d::Zero};
+
+    /// \brief Topic name to subscribe
+    public: std::string topicName{""};
+
+    /// \brief List of topics publishing LaserScan messages.
+    public: QStringList topicList;
+
+    /// \brief Entity representing the sensor in the world
+    public: sim::Entity frustumEntity;
+
+    /// \brief Mutex for variable mutated by the checkbox and spinboxes
+    /// callbacks.
+    /// The variables are: msg
+    public: std::mutex serviceMutex;
+
+    /// \brief Initialization flag
+    public: bool initialized{false};
+
+    /// \brief Reset visual flag
+    public: bool resetVisual{false};
+
+    /// \brief frustum visual display dirty flag
+    public: bool visualDirty{false};
+
+    /// \brief frustum sensor entity dirty flag
+    public: bool frustumEntityDirty{true};
+  };
+}
+}
+}
+
+using namespace gz;
+using namespace sim;
+
+/////////////////////////////////////////////////
+VisualizeFrustum::VisualizeFrustum()
+  : GuiSystem(), dataPtr(new VisualizeFrustumPrivate)
+{
+  // no ops
+}
+
+/////////////////////////////////////////////////
+VisualizeFrustum::~VisualizeFrustum()
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->serviceMutex);
+  this->dataPtr->scene->DestroyVisual(this->dataPtr->frustum);
+}
+
+/////////////////////////////////////////////////
+void VisualizeFrustum::LoadFrustum()
+{
+  auto loadedEngNames = rendering::loadedEngines();
+  if (loadedEngNames.empty())
+    return;
+
+  // assume there is only one engine loaded
+  auto engineName = loadedEngNames[0];
+  if (loadedEngNames.size() > 1)
+  {
+    gzdbg << "More than one engine is available. "
+          << "VisualizeFrustum plugin will use engine ["
+          << engineName << "]" << std::endl;
+  }
+  auto engine = rendering::engine(engineName);
+  if (!engine)
+  {
+    gzerr << "Internal error: failed to load engine [" << engineName
+          << "]. VisualizeFrustum plugin won't work." << std::endl;
+    return;
+  }
+
+  if (engine->SceneCount() == 0)
+    return;
+
+  // assume there is only one scene
+  // load scene
+  auto scene = engine->SceneByIndex(0);
+  if (!scene)
+  {
+    gzerr << "Internal error: scene is null." << std::endl;
+    return;
+  }
+
+  if (!scene->IsInitialized() || scene->VisualCount() == 0)
+  {
+    return;
+  }
+
+  // Create frustum visual
+  gzdbg << "Creating frustum visual" << std::endl;
+  auto root = scene->RootVisual();
+
+  this->dataPtr->frustum = scene->CreateFrustumVisual();
```

Comment:
The [backport PR](https://github.com/gazebosim/gz-rendering/pull/1102) in gz-rendering changed this function call in order to be ABI compatible. Can you change this call to the following with the TODO note as a reminder for me to update this in the future?

```suggestion
  // Create a frustum visual
  // \todo(iche033) uncomment and use official API in gz-rendering10
  // this->dataPtr->frustum = scene->CreateFrustumVisual();
    this->dataPtr->frustum =
      std::dynamic_pointer_cast<rendering::FrustumVisual>(
      scene->Extension()->CreateExt("frustum_visual"));
```